src/App.js

```diff
@@ -277,9 +295,27 @@ const StartedRoute = ({component: Component, ...rest}) => (
  */
 const returnDefaultComponent = (Component, props) => {
   if (version.checkWalletVersion()) {
-    return (
-      <div className="component-div"><Navigation {...props}/><Component {...props} /><RequestErrorModal {...props} /></div>
-    );
+    if (props.location.pathname === '/locked/' && hathorLib.wallet.isHardwareWallet()) {
```

Comment:
Why are we doing it here? It seems that the previous version of this code was way simpler.

Reply:
We need to add this `if` because we don't have a lock screen for hardware wallet, if the screen is locked, we clean the wallet data and redirect to the initial screen.

The other new code is the alert modal and there is a big comment there explaining why it should be here

Reply:
Makes sense. Thanks!